src/engine/client/client.cpp

```diff
@@ -2431,18 +2431,25 @@ void CClient::HandleTeeworldsConnectLink(const char *pConLink)
 int main(int argc, const char **argv) // ignore_convention
 {
 #if defined(CONF_FAMILY_WINDOWS)
+	bool UseDefaultConsoleSettings = true;
 	for(int i = 1; i < argc; i++) // ignore_convention
 	{
 #ifdef CONF_RELEASE
-		if(!(str_comp("-c", argv[i]) == 0 || str_comp("--console", argv[i]) == 0)) // ignore_convention
+		if(str_comp("-c", argv[i]) == 0 || str_comp("--console", argv[i]) == 0) // ignore_convention
 #else
 		if(str_comp("-s", argv[i]) == 0 || str_comp("--silent", argv[i]) == 0) // ignore_convention
 #endif
 		{
-			FreeConsole();
+			UseDefaultConsoleSettings = false;
 			break;
 		}
 	}
+#ifdef CONF_RELEASE
+	if(!UseDefaultConsoleSettings)
```

Comment:
> I think this is inverted.

Yeah, you're right :( cf #1599 

> To improve readability, I suggest to use a variable HideConsole set to true for release and false otherwise.

Eh, but then you have to put #ifdef to know if you set it to true or false. It bloats the code a bit.

Reply:
> your solution uses #ifdef twice, I suggest to use it only once for initialisation

I'm a bit confused. Since it's a short bit of code, maybe you could sumbit a PR? :)

Reply:
i cant right know but I'll give a sketch:
```
#ifdef release
bool Hide = true
#else
bool Hide = false
#endif
for arg in argv {
  if arg == -c
    Hide = false
  if arg == -s
    Hide = true
if Hide
  FreeConsole
```

Reply:
@nheir Oh, good :) Just need a break after each `if`